programs/eosio-launcher/main.cpp

```diff
@@ -1650,14 +1651,30 @@ launcher_def::bounce (const string& node_numbers) {
       const host_def& host = node_pair.first;
       const eosd_def& node = node_pair.second;
       string node_num = node.name.substr( node.name.length() - 2 );
-      string cmd = "cd " + host.eosio_home + "; "
-                 + "export EOSIO_HOME=" + host.eosio_home + string("; ")
-                 + "export EOSIO_NODE=" + node_num + "; "
-                 + "./scripts/eosio-tn_bounce.sh " + eosd_extra_args;
       cout << "Bouncing " << node.name << endl;
-      if (!do_ssh(cmd, host.host_name)) {
-         cerr << "Unable to bounce " << node.name << endl;
-         exit (-1);
+      if (!host.is_local()) {
+         string cmd = "cd " + host.eosio_home + "; "
+                    + "export EOSIO_HOME=" + host.eosio_home + string("; ")
+                    + "export EOSIO_NODE=" + node_num + "; "
+                    + "./scripts/eosio-tn_bounce.sh " + eosd_extra_args;
+         if (!do_ssh(cmd, host.host_name)) {
+            cerr << "Unable to bounce " << node.name << endl;
+            exit (-1);
+         }
+      }
+      else {
+         string cmd = "./scripts/eosio-tn_bounce.sh " + eosd_extra_args;
+         bp::child c(cmd,
+                     bp::env["EOSIO_HOME"] = host.eosio_home,
+                     bp::env["EOSIO_NODE"] = node_num );
+
+         if(!c.running()) {
+           cerr << "child not running after spawn " << cmd << endl;
+           for (int i = 0; i > 0; i++) {
```

Comment:
We really want this tight loop, with no delay, running till i rolls over to negative?

Reply:
A lot of this code is copy-pasted from elsewhere in the launcher. I will simplify it and pull it out into a function.